packages/remix-react/components.tsx

```diff
@@ -1424,17 +1428,16 @@ export const LiveReload =
             console.log("Remix dev asset server web socket error:");
             console.error(error);
           };
-        }).toString();
-
-        return (
-          <script
-            nonce={nonce}
-            suppressHydrationWarning
-            dangerouslySetInnerHTML={{
-              __html: `(${setupLiveReload})(${port})`,
-            }}
-          />
-        );
+        };
+
+        React.useEffect(() => {
+          if (!liveReloadMounted) {
+            setupLiveReload(port);
+            liveReloadMounted = true;
+          }
```

Comment:
Instead of this, why don't we put the body of `setupLiveReload` in this effect and do proper cleanup? That way we don't need to have the mutable variable.

Reply:
I'd rather have the mutable global here than re-connect to a port if a boundary is hit personally. This should only ever happen once and stay connected for the duration of the session.

Reply:
Yeah, good point 👍